### DISCUSSION TITLE
Saving odf

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#27] [General]
The software does not query overwritting an existing odf by same name and neither does it allow one to choose where to save.

Save... and Save as... options should be available.

**Comment by Bumblebee001** [comment]
OrganBuilder by Vincent Foreman allows the software to save work as a separate file with its own extension..... the odf is created as an "export project for GrandOrgue" option and this is when the actual odf is generated. I would have simply called the latter option "Create ODF". Here to one is allowed to select where to save this odf.

I think there could be an argument for saving the "project" as a separate entity from the actual odf perhaps....

**Reply by larspalo** [reply]
Sorry, by design there's no internal file representation of the odf. GOODF just work with the .organ file as it is. Reading the file into the program when opening, saving the file by writing it back to disc. To "Save as..." you simply alter the "Name of .organ file (without extension)" on the [Organ] panel.

And GOODF should warn you the "first time" you try to overwrite a file. But not the second time! This is also by design to minimize the "nagging". You can read what other users think about it here https://github.com/GrandOrgue/grandorgue/discussions/1394#discussioncomment-5580376.

**Comment by Bumblebee001** [comment]
As I pointed out in another post/ email, there seems to be a bug. I had to almost re-do the entire odf because I could not check which manual was the Pedals only to find out that it is the first entry that assumes and allows it but not the others. Why???

This error on loading the organ keeps being generated: Stop 16 (Windchest 0)..... when I did not enter any such information!!! It is generated by itself as it were.

27/05/2023 02:26:29: 02:26:29: Error: Out of range value at section 'Stop016' entry 'WindchestGroup': 0

I have to go in with Notepad++ and manually remove this odd mysterious entry. A Stop017 is also present.

Setting graphics in terms of x and y co=-ordinates will remain a "PITA" (excuse the expression). jOrgan handles this so much better and should be emulated..... please do........ You can move objects and size them whilst you can see what you have and where you're going. x/y is pure guess work or you have to keep a record, calculating from size of the images and you can predict the next placement but it's always been such a hassle.

Why is Dark Red the default colour???? Please make default Black.

[WindchestGroup001]
Name=1 Pedal
NumberOfEnclosures=0
NumberOfTremulants=0

These are not registering correctly!

Mark

**Reply by Bumblebee001** [reply]
Also....additional details added at the default starting screen are not saved

**Reply by Bumblebee001** [reply]
It would help to define x/y size of each panel. Positioning items is difficult unless one knows the height and width in actual figures. 

How can one find say the co-ordinates of the centre of a "Medium" panel, or "Medium Large"? :-/

**Reply by Bumblebee001** [reply]
Oooo. Now I see the customizable option..... coool! 

Defining pre-given options remains desirable.

**Reply by Bumblebee001** [reply]
The way I ended up building the odf because I started with the HW section instead of the pedal and having had to reassemble the HW after the Positif, the keyboards are reversed. Can we have options that are less restrictive so we can start where we want and build up gradually as takes our fancy and be able to order the keyboards in different order to our preference?

Even in reality, some organs have the Great at the bottom, others in the middle...... You may wish to have the upper of 2 keyboards as the great as inadvertently I got with the positif at the bottom..... or the otherway round with the positif/ swell at the top, above the great....... You get the idea....

**Reply by larspalo** [reply]
> I had to almost re-do the entire odf because I could not check which manual was the Pedals only to find out that it is the first entry that assumes and allows it but not the others. Why???

Because that's what GO wants (read the third sentence in the help section for "Manual objects" in GO help). Also, it's because of how GO enumerates the manuals to indicate a pedal ([Manual000]). But this gets even more interesting and complicated. With the new panel format you _can_ actually decide to display any manual as the pedal by specifying that the panel "HasPedals" and adding that manual as the first manual entry (on that panel) and then in the order you want them displayed (from bottom and upwards).

And to make things even more complicated, if you use manual x/y pixel positioning and custom bitmaps: _anything_ can be displayed as _whatever and however_ you want it... But because of GO and it's history again: _"Best practice is to specify the visible manuals in the order of appearance, lowest first."_ This is also noted in the GOODF help.

So, as you can see in the screenshot below. I **"can"** add the manuals in whichever order I want and completely ignore ticking the "This is the Pedal" checkbox", and still display the manuals on the panel as I want them (_look carefully at the order I've added them_). But the fact that I _"can"_ do it that way is not to say that I _"should"_ do it that way. Rather, one **shouldn't** do it like I've done on purpose here!

![Skärmbild från 2023-05-27 08-57-12](https://github.com/larspalo/GOODF/assets/6173941/5e6184de-b6a2-499e-a21c-1cd0142b6974)

"_Best practice is to specify the visible manuals in the order of appearance, lowest first._"

**Reply by larspalo** [reply]
> This error on loading the organ keeps being generated: Stop 16 (Windchest 0)..... when I did not enter any such information!!! It is generated by itself as it were.

This is caused by not specifying what windchest the stop/rank should be on. You've just forgotten to select any windchest, no worries, just select one.

**Reply by larspalo** [reply]
> Why is Dark Red the default colour???? Please make default Black.

Because that's the default color GO will use if none other is specified.

**Reply by larspalo** [reply]
> It would help to define x/y size of each panel. Positioning items is difficult unless one knows the height and width in actual figures.

The size of the panel can be found in the "Displaymetrics" of that panel.

> How can one find say the co-ordinates of the centre of a "Medium" panel, or "Medium Large"? :-/

On the "Displaymetrics" panel window - the sizes for the text entered sizes are still shown grayed out in the spinctrls.

To find the centre, divide the (grayed out) panel width value by 2, and then deduct half of the width value of the object you're adding. That gives you the x coordinate to specify for that object. A "PITA"? Maybe, but for now that's how we must deal with it.

Personally, I tend to use GIMP (image editing software) to create the graphics and find placements by using multiple layers I can move around. I also usually have GO ready with the sample set opened and re-load as I change the different values to see the results... Until I've found the time to implement a panel rendering in GOODF that exactly does what GO does - this is at least a working solution!
